pkg/datapath/garp/cells.go

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package garp
+
+import (
+	"github.com/spf13/pflag"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	"github.com/cilium/cilium/pkg/k8s/utils"
+	nodeTypes "github.com/cilium/cilium/pkg/node/types"
+)
+
+const (
+	// L2PodAnnouncementsInterface is the interface used to send Gratuitous ARP messages.
+	L2PodAnnouncementsInterface = "l2-pod-announcements-interface"
+
+	EnableL2PodAnnouncements = "enable-l2-pod-announcements"
+)
+
+// Config contains the configuration for the GARP cell.
+type Config struct {
+	L2PodAnnouncementsInterface string
+	EnableL2PodAnnouncements    bool
+}
+
+func (def Config) Flags(flags *pflag.FlagSet) {
+	flags.String(L2PodAnnouncementsInterface, def.L2PodAnnouncementsInterface, "Interface used for sending gratuitous arp messages")
+	flags.Bool(EnableL2PodAnnouncements, def.EnableL2PodAnnouncements, "Enable announcing Pod IPs with Gratuitous ARP")
+}
+
+var localPodsCell = cell.ProvidePrivate(func(lc hive.Lifecycle, c k8sClient.Clientset) (resource.Resource[*corev1.Pod], error) {
```

Comment:
I'm really sorry about this, but we now have `LocalPodResource` available from `daemon/k8s/resources.go`, so this can now be dropped in favor of it. The only changes you need are removing this and changing `resource.Resource[*corev1.Pod]` into `k8s.LocalPodResource` (package is `daemon/k8s`). Note that it's `slim_corev1.Pod` now. Hope it contains everything you need. 

I'm still in the process of switching `pkg/k8s/watchers/pod.go` to use `LocalPodResource`, but should be done with that soon after which we should (in the normal use when we don't watch all pods) have only a single pod watcher in the whole agent.